### ISSUE TITLE
Separate the `LicenseClassifications` model for (de-)serialization from the API for programmatic use

### ISSUE THREAD

**Issue by sschuberth** [oss-review-toolkit/ort#6191]
I was considering that, too, and played with the idea. While doing so I realized several things:
- The "API" for `LicenseClassifications` is terrible, see e.g. the need for all the lazy properties. That's mostly because `LicenseClassifications` is a model class for (de-)serialization, and it should be used for that purpose only. We'd actually need a wrapping / companion class that defines a nice API for license classifications for programmatic use. This is similar to the ORT result interface that we discussed.
- The serialization could use sets sorted by `id` (for the classifications) and by `name` (for the categories). I guess we initially chose lists (with manual checks for duplicates) as we saw some value in maintaining a human-created order in these files. But I'm not convinced anymore that the order needs to be maintained when rewriting these files.

So, long story short, I'd prefer to address this in a follow-up PR on the `LicenseClassifications` level in a general way rather than sorting here.

_Originally posted by @sschuberth in https://github.com/oss-review-toolkit/ort/pull/6180#discussion_r1042573136_
